Lib/StringUtils.cpp

```diff
@@ -45,6 +45,17 @@ vstring StringUtils::replaceChar(vstring str, char src, char target)
   return vstring(buf.array(), len);
 }
 
+void StringUtils::replaceAll(vstring& where, const vstring& from, const vstring& to)
```

Comment:
Might be worth warning about quadratic runtime in a comment.

Reply:
The next replacement can only happens after a position where the previous ended. Is it still quadratic?

Although if you do `a^100,a,b^100`, you will indeed get an answer `b^10000`. But isn't that what you expect from a replaceAll?

Reply:
So from my quick reading I'd expect `where.replace` to have linear-ish time, and so the thing has complexity `O(mn)` where `m` is the length of `where` and `n` is the number of occurrences of `from`. But `n` is very likely small in practice so probably not a big deal, you're right.